### ISSUE TITLE
Feature Request: Option to ignore nested functions

### ISSUE THREAD

**Issue by torfsen** [econchick/interrogate#11]
**Is this a BUG REPORT or FEATURE REQUEST?**:

* [ ] bug
* [x] feature

## Description of Bug or Feature

Nested functions (i.e. function definitions within function definitions) are usually small helpers that need no docstring and are not part of the public API. It should therefore be possible to optionally ignore them.

**Comment by econchick** [comment]
Thanks for the request! Would the `-r/--ignore-regex` help in this case? For instance:

```py
# sample.py
"""Sample module-level docs"""

def top_level_func():
    """A top level function"""

    def inner_func():
        pass
```

```console
$ interrogate -r inner_func tests/functional/sample/full.py -vv

=== Coverage for /Users/lynn/dev/interrogate/tests/functional/sample/ ====
--------------------------- Detailed Coverage ----------------------------
| Name                                      |                     Status |
|-------------------------------------------|----------------------------|
| full.py (module)                          |                    COVERED |
|   top_level_func (L5)                     |                    COVERED |
|-------------------------------------------|----------------------------|

-------------------------------- Summary ---------------------------------
| Name           |       Total |       Miss |       Cover |       Cover% |
|----------------|-------------|------------|-------------|--------------|
| full.py        |           2 |          0 |           2 |         100% |
|----------------|-------------|------------|-------------|--------------|
| TOTAL          |           2 |          0 |           2 |       100.0% |
------------ RESULT: PASSED (minimum: 80.0%, actual: 100.0%) -------------
```

**Comment by torfsen** [comment]
> Would the `-r/--ignore-regex` help in this case?

In theory, yes -- but that would require me to hardcode all of the inner functions, since they usually don't follow any systematic naming scheme. That's something I'd like to avoid when working on a larger code base. Ignoring the nested functions completely (just like, say, magic functions) would be much easier.

It's not something that's a deal breaker for me, but it would be a welcome addition.

**Comment by econchick** [comment]
Ah, understood - I'll see if I can figure something out. Thanks for the idea!

**Comment by econchick** [comment]
Hey @torfsen - I've added this feature for ya in #24 via a new flag: `-n` / `--ignore-nested-functions`. It will be released in v1.2.0, but I want to try and address your other feature request before actually making that release. Feel free to try it out before then via 
```
pip install git+https://github.com/econchick/interrogate.git@cd06c57b91cf8adf6fe692d310cdb1cb4afb1640
``` 

I'll update this issue once it is released on PyPI either way.

**Comment by econchick** [comment]
I've released 1.2.0 with this feature included - thank you for the suggestion!
